### ISSUE TITLE
NavigationView Navigate throw System.NullReferenceException

### ISSUE THREAD

**Issue by Porterbg** [lepoco/wpfui#800]
### Describe the bug

Sample code 
```
MainWindow _mainWindow = App.GetService<MainWindow>() as MainWindow;
_mainWindow.navView.Navigate(typeof(SomePage));
```
Call Stack: Wpf.Ui.Controls.Navigate->NavigateInternal-.UpdateContent

### To Reproduce

Just try to use Navige function of NavigationView

### Expected behavior

Should not throw exception

### OS version

windows 10

### .NET version

.net 6

### WPF-UI NuGet version

3.0.0-preview.7 but it is the same in previous 3.0.0 previews

**Comment by anesdiaf** [comment]
Hi,

1. You should learn how .Net Dependency injection with WPF works.
2. If you go to app.xaml you'll see that the service for "MainWindow" is add as an implementation of "INavigationWindow" 
`services.AddScoped<INavigationWindow, MainWindow>();`
That being said if you want to get an implementation of a service you'll have to get it through a Factory pattern (so you can use it for other Windows which gives you added scalability),
or simply 
`MainWindow saleWindow = App.GetService<INavigationWindow>() as MainWindow;`

Hope i've helped.

**Comment by Porterbg** [comment]
I was looking at Gallery example where implementation was with services.AddSingleton<IWindow, MainWindow>();

**Comment by anesdiaf** [comment]
My bad, i'm still using the 2.1 version, 

but after going through the code of 3.0 preview version, 
I found that this may not be a bug,
because the IWindow or the MainWindow or NavigationView them selves doesn't have a navigation function,
rather the MainWindow (you can see in it's constructor) is Subscribing to navigation changes through an injected navigation service,

`public MainWindow(
        MainWindowViewModel viewModel,
        ->INavigationService navigationService,<-
        IServiceProvider serviceProvider,
        ISnackbarService snackbarService,
        IContentDialogService contentDialogService
    )`

this service then wraps the NavigationView control (navView in your case)

`navigationService.SetNavigationControl(NavigationView);`

then NavigationView control uses this function to retrieve page instances and view models

`NavigationView.SetServiceProvider(serviceProvider);`

So, to navigate you need this navigation service either by injecting to a constructor or using the service provider to get it
`_navigationService.Navigate(typeof(SomePage));`

My advice is still on you need to learn how ioc di works.

Thanks.

**Comment by Porterbg** [comment]
I still have a problem but I'm not sure if it is possible.
I need to user UserControl as I want to use the PopUp from Hardcodet.NotifyIcon. In it I want to use NavigationView and to be able to navigate to specific Item. I can't inject in the UserControls constructor so right after creating the PopUp I do
```
navigationService.SetNavigationControl(nvSample);
nvSample.SetPageService(pageService);
_navigationService= navigationService;
```
then I call `_navigationService.Navigate(pageType);` but I get `System.NullReferenceException: 'Object reference not set to an instance of an object.'
` as _pageService in _navigationService is null. Otherwise if I don't navigate from code the PopUp works fine.

**Comment by MakesYT** [comment]
try this
```
        NavigationView.Loaded += (_, _) =>
        {
            var navigationService = ServiceManager.Services.GetService<INavigationService>()!;

            navigationService.SetNavigationControl(NavigationView);
            NavigationView.SetServiceProvider(ServiceManager.Services);
        };
```

**Comment by Porterbg** [comment]
Tried
```
nvSample.Loaded += (_, _) =>
            {
                var navigationService = App.GetService<INavigationService>();
                navigationService.SetNavigationControl(nvSample);
                nvSample.SetServiceProvider(App.GetService<IServiceProvider>());
                _navigationService = navigationService;
            };
```
It does not work.

**Comment by MakesYT** [comment]
> Tried 试图
> 
> ```
> nvSample.Loaded += (_, _) =>
>             {
>                 var navigationService = App.GetService<INavigationService>();
>                 navigationService.SetNavigationControl(nvSample);
>                 nvSample.SetServiceProvider(App.GetService<IServiceProvider>());
>                 _navigationService = navigationService;
>             };
> ```
> 
> It does not work.它不工作。

Are you sure you registered the corresponding page in the services

**Comment by MakesYT** [comment]
> Tried
> 
> ```
> nvSample.Loaded += (_, _) =>
>             {
>                 var navigationService = App.GetService<INavigationService>();
>                 navigationService.SetNavigationControl(nvSample);
>                 nvSample.SetServiceProvider(App.GetService<IServiceProvider>());
>                 _navigationService = navigationService;
>             };
> ```
> 
> It does not work.

Are you sure you registered the corresponding page in the services

**Comment by Porterbg** [comment]
Yes I did and it is not the pages that cause the problem. It seems that there is no page service.

**Comment by MakesYT** [comment]
> Yes I did and it is not the pages that cause the problem. It seems that there is no page service.

Oh, I might know why SetServiceProvider did you get a new ServiceProvider? Not the one you built in the first place?

**Comment by Porterbg** [comment]
> > Yes I did and it is not the pages that cause the problem. It seems that there is no page service.
> 
> Oh, I might know why. SetServiceProvider did you get a new ServiceProvider? Not the one you built in the first place?

No, I can't change the constructor of the popup so how could I get a new one?
